sopel/config/core_section.py

```diff
@@ -578,16 +578,19 @@ def homedir(self):
     """A list of hostnames which Sopel should ignore.
 
     Messages from any user whose connection hostname matches one of these
-    values will be ignored. :ref:`Regular expression syntax <re-syntax>` is
-    supported:
+    values will be ignored. Both :ref:`regular expression syntax <re-syntax>`
+    and exact match are supported:
 
     .. code-block:: ini
 
         host_blocks =
-            domain\\.com
-            .*\\.domain\\.com
+            exact-match.com
+            .*\\.regex-domain\\.com
```

Comment:
There is an ``or`` though, that uses exact match. What would be the correct phrasing for that?

Reply:
It's not a simple matter, is it? Technically, both regex and exact-match _are_ supported. The issue is with managing the user's expectations (with a side of "why does the code try to treat the same value two ways?!", because technical debt).

The possibility for overlap between exact-match and regex syntax is probably greater for nicks than hosts (due to brackets/braces being allowed in nicknames), but it theoretically exists for both.

Maybe it would be enough to mitigate the possibility for confusion by attempting an exact match first, and only trying the regex path if that fails? Then a naïve user accidentally using regex syntax in what they think is an exact match won't be confused that their block rule didn't work.

On the other hand, trying exact matches first has its own potential for accidental false-positives on the nickname side… But users who choose regex-like nicknames deserve to be blocked anyway. 😛

Reply:
I see that there is a problem, but I'm not sure what you want me to do about it in this PR.

Since I'm a bit confused, the safe path for me is to revert to a more elusive explanation for now. Something like: "Regular expression are supported (remember to escape characters as needed)."?

I could also add a little warning, something like "We are working on a better block system, see issue [link] for more information and update". I feel like this approach is honest with where the code is at the moment (as you mentioned: technical debt), and I believe honest communication will help the user understand why it isn't that simple.

What do you say?

Reply:
![Screen Shot 2020-05-22 at 17 05 07](https://user-images.githubusercontent.com/3259552/82681731-7acf2780-9c4e-11ea-8f36-d283c40dd889.png)

Maybe it could be a "note" instead of a warning, but I wanted you to see how it looks like first.

Reply:
That screenshot looks good, though I agree about using `.. note::` instead.

I assume this is for both `host_blocks` and `nick_blocks`? Maybe the `.. warning::` would be the bit about regular expression syntax—but that might be too many blocks. 😅

Reply:
Done!